hazelcast/src/main/java/com/hazelcast/nio/SerializationClassNameFilter.java

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Implementation of basic protection against untrusted deserialization. It holds blacklist and whitelist with classnames and
+ * package names.
+ *
+ * @see #filter(String)
+ */
+public final class SerializationClassNameFilter implements ClassNameFilter {
+
+    /**
+     * The default checker with a blacklist containing some well-known vulnerable classes/packages.
+     */
+    public static final SerializationClassNameFilter DEFAULT_CLASSNAME_FILTER = createDefaultChecker();
+
+    private static final String DESERIALIZATION_ERROR = "Resolving class %s is not allowed.";
+
+    private final ShadeOfGreyList blacklist;
+    private final ShadeOfGreyList whitelist;
+
+    public SerializationClassNameFilter(JavaSerializationFilterConfig config) {
+        if (config == null) {
+            throw new IllegalArgumentException("JavaSerializationFilterConfig has to be provided");
+        }
+        blacklist = new ShadeOfGreyList(config.getBlacklist());
+        whitelist = new ShadeOfGreyList(config.getWhitelist());
+    }
+
+    /**
+     * Throws {@link SecurityException} if the given class name appears on the blacklist or does not appear on a non-empty
+     * whitelist.
+     *
+     * @param className class name to check
+     * @throws SecurityException if the classname is not allowed for deserialization
+     */
+    public void filter(String className) throws SecurityException {
+        if (blacklist.isListed(className) || (!whitelist.isEmpty() && !whitelist.isListed(className))) {
+            throw new SecurityException(format(DESERIALIZATION_ERROR, className));
+        }
+    }
+
+    /**
+     * Black/WhiteList holder.
+     */
+    private static class ShadeOfGreyList {
+        private final Set<String> classes;
+        private final Set<String> packages;
+
+        ShadeOfGreyList(ClassFilterList configList) {
+            classes = configList.getClasses().isEmpty() ? null : new HashSet<String>(configList.getClasses());
+            packages = configList.getPackages().isEmpty() ? null : new HashSet<String>(configList.getPackages());
+        }
+
+        boolean isListed(String className) {
+            if (classes != null && classes.contains(className)) {
+                return true;
+            }
+            if (packages != null) {
+                int dotPosition = className.lastIndexOf(".");
+                if (dotPosition > 0) {
+                    String packageName = className.substring(0, dotPosition);
+                    return packages.contains(packageName);
+                }
+            }
+            return false;
+        }
+
+        boolean isEmpty() {
+            return classes == null && packages == null;
+        }
+    }
+
+    /**
+     * Creates default checker with a blacklist containing some well-known vulnerable classes/packages.
+     *
+     * @return checker instance with a default blacklist
+     */
+    private static SerializationClassNameFilter createDefaultChecker() {
```

Comment:
The content of the default configuration should be in the com.hazelcast.config. Now it is hidden somewhere in the system.